### ISSUE TITLE
Reprocess older workout weather data

### ISSUE THREAD

**Issue by Dryusdan** [SamR1/FitTrackee#480]
Hello 👋 

First, thank you for this amaizing project, I can now download all of my bike data without any dependent service. 
Actually, this issue is not an issue but a question.

I fetch and download all of my bike workout from Garmin (I made a little program, may I tall about it later (it's open source but in earlier state)) and I generate too many requests on visualcrossing. 
My account was "banned" from visualcrossing for overcharge after multiple HTTP code 429, I receive 401. I can "unblock" my account, but all of my acitivies sended haven't weather data.

Is there a way to reprocess all workout without weather data ? :) 

Thank by advance :)

**Comment by SamR1** [comment]
Hi,

Thanks for your feedback.

For now it is not possible via the API to reprocess workouts to get weather data or recalculate some values like max speed for instance (only [few values](https://samr1.github.io/FitTrackee/en/api/workouts.html#patch--api-workouts-(string-workout_short_id)) can be updated). 

Maybe one workaround is to add a pause between workouts imports to avoid exceeding Weather API rate limit.

**Comment by Dryusdan** [comment]
Hum, okay.

Okay, I will add a rate limit for import in case of weather is enable on fittrackee instance. Thank :)

**Comment by SamR1** [comment]
is there a lot of workouts to reprocess?

**Comment by Dryusdan** [comment]
Approximately 1400 ^^ (I forgot to remove my weather key on my dev instance 😅 )

**Comment by SamR1** [comment]
yes, for bulk imports via scripts, the other solution is to temporarily remove the key.
